test/util/testfactory/utils.go

```diff
@@ -102,3 +114,211 @@ func GenerateAccounts(count int) []string {
 	}
 	return accs
 }
+
+func NewBaseAccount(kr keyring.Keyring, name string) (*authtypes.BaseAccount, sdk.Coins) {
+	if name == "" {
+		name = tmrand.Str(6)
+	}
+	rec, _, err := kr.NewMnemonic(name, keyring.English, "", "", hd.Secp256k1)
+	if err != nil {
+		panic(err)
+	}
+	addr, err := rec.GetAddress()
+	if err != nil {
+		panic(err)
+	}
+	origCoins := sdk.Coins{sdk.NewInt64Coin(bondDenom, BaseAccountDefaultBalance)}
+	bacc := authtypes.NewBaseAccountWithAddress(addr)
+	return bacc, origCoins
+}
+
+func GenerateBaseAccounts(kr keyring.Keyring, names []string, initExtraCoins ...sdk.Coin) ([]authtypes.BaseAccount, []banktypes.Balance) {
+	bAccounts := make([]authtypes.BaseAccount, len(names))
+	balances := make([]banktypes.Balance, len(names))
+	for i, name := range names {
+		acc, coins := NewBaseAccount(kr, name)
+		bAccounts[i] = *acc
+		balances[i] = banktypes.Balance{
+			Address: acc.GetAddress().String(),
+			Coins:   coins.Add(initExtraCoins...),
+		}
+	}
+
+	return bAccounts, balances
+}
+
+func GenerateDelayedVestingAccounts(kr keyring.Keyring, names []string, initUnlockedCoins ...sdk.Coin) ([]*vestingtypes.DelayedVestingAccount, []banktypes.Balance) {
```

Comment:
I would prefer if we have a comment explaining how the delayed vesting accounts are created. Something like:

```suggestion
// GenerateDelayedVestingAccounts generates a set of delayed vesting accounts with the corresponding init unlocked coins.
// The end time of the vesting accounts starts from the past, i.e. -2sec, for the first name,
// and keeps increasing with every account at a rate of 5sec per account.
func GenerateDelayedVestingAccounts(kr keyring.Keyring, names []string, initUnlockedCoins ...sdk.Coin) ([]*vestingtypes.DelayedVestingAccount, []banktypes.Balance) {
```

Reply:
mark resolved as the code does not exist anymore